### PR TITLE
base64 image

### PR DESCRIPTION
**Pull request by stephane-f**

### CODE REVIEW

File: Source/Widgets/CabbageWidgetData.cpp [hunk 1]
```diff
@@ -310,6 +310,7 @@ void CabbageWidgetData::setCustomWidgetState (ValueTree widgetData, String lineO
         switch (HashStringToInt (identifier.toStdString().c_str()))
         {
             //======== strings ===============================
+            case HashStringToInt ("base64"):
```
Comment (rorywalsh) [inline Source/Widgets/CabbageWidgetData.cpp:313]: I'm not sure we need a new identifier here. I think it is better to add base64 support to the file identifier as we discussed on the forum?

File: Source/Widgets/CabbageWidgetDataInitMethods.cpp [hunk 2]
```diff
@@ -866,6 +866,7 @@ void CabbageWidgetData::setVSliderProperties (ValueTree widgetData, int ID)
 void CabbageWidgetData::setImageProperties (ValueTree widgetData, int ID)
 {
     setProperty (widgetData, "basetype", "layout");
+    setProperty (widgetData, CabbageIdentifierIds::base64, "");
```
Comment (rorywalsh) [inline Source/Widgets/CabbageWidgetDataInitMethods.cpp:869]: Again, I don't think we need this identifier..

File: Source/Widgets/CabbageImage.cpp [hunk 3]
```diff
@@ -155,21 +157,19 @@ void CabbageImage::valueTreePropertyChanged (ValueTree& valueTree, const Identif
...
+    if(imgBase64.isNotEmpty())
+    {
+        MemoryOutputStream out;
+        bool result = Base64::convertFromBase64(out, imgBase64);
+        if(result)
+            img = ImageFileFormat::loadFrom(out.getData(), out.getDataSize());
+    }
+    else 
+    {
+		imgFile = File(CabbageWidgetData::getStringProp(valueTree, CabbageIdentifierIds::csdfile)).getParentDirectory().getChildFile(CabbageWidgetData::getStringProp(valueTree, CabbageIdentifierIds::file));
```
Comment (rorywalsh) [inline Source/Widgets/CabbageImage.cpp:170]: This block will only be hit if the file() identifier returns an empty string, in which case it makes no sense in trying to load an image. We need to first check if the file string is not empty, and then proceed to validate it as either a base64 string, or a file.

File: Source/Widgets/CabbageWidgetData.cpp (hunk 1 above)
Comment (stephane-f) [reply Source/Widgets/CabbageWidgetData.cpp:313]: You're right. Your approach is better than mine as we discussed on the forum

**Comment by rorywalsh** [comment]
We have resolved these issues via discussions on the forum, so I'm going to close this PR. Thanks for the contributions, they are always welcome ;)

**Pull request closed without merging**